packages/juggler/src/model.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Options, AnyType} from './common';
+import {Type} from './types';
+
+/**
+ * This module defines the key classes representing building blocks for Domain
+ * Driven Design.
+ * See https://en.wikipedia.org/wiki/Domain-driven_design#Building_blocks
+ */
+
+/**
+ * Property definition for a model
+ */
+export class ModelProperty {
+  name: string;
+  type: string | Function | Object | Type<AnyType>; // For example, 'string', String, or {}
+  json?: PropertyForm;
+  store?: PropertyForm;
```

Comment:
These two config options `json` and `store` look unfamiliar to me. Are they already recognized and implemented in loopback-datasource-juggler? If not, then I propose to remove them from this patch in order to keep the patch focused on providing the existing juggler API, and to make it easier in the future to understand what code was added in order to support `json` and `store` options.